### DISCUSSION TITLE
How to prompt for input?

### DISCUSSION THREAD

**Discussion by lostecmo** [alagrede/znote-app#55] [Q&A] [answered]
Wondering if there is any way to prompt and get input from the user in notebookjs.  It would be useful for prototyping some small scripts that I write.  Maybe create an HTML input form?  Use Node to read from the console (e.g. readline)?

**Comment by alagrede** [comment, marked as answer]
Hi @lostecmo,
Thank you for your feedback 😊
I don't know if this is enough for your use case, but we already have **2 similar features** to get user input.
But I will think about displaying an HTML input, it would be prettier 👍

## 1 - The **load block**:

**Declaration**
\`\`\`json block1
{ "name": "val1", "props": "val2"}
\`\`\`

**Usage**
\`\`\`js
print(loadBlock('block1'));
\`\`\`

## 2 - The **global block**:

\`\`\`js
//global 1
const myVar = "Hello";
\`\`\`

\`\`\`js
print(myVar);
\`\`\`

I hope this will help you.
Anthony

**Comment by lostecmo** [comment]
As a follow-up, a few more examples of using the Display HTML element (htmlEL and el) would be helpful as well.

Thanks,
Alan
________________________________

**Reply by alagrede** [reply]
👍 Ok I will do that. I also update this thread when I could show an HTML input.
Thanks

**Comment by lostecmo** [comment]
Thanks for the quick response.  To be honest, I am probably looking for more than is currently possible.  I would like to be able to run my code block and have a prompt (HTML for or even some type of console) to get input.  I can at least work on most of my script logic with things as they are.  The load block will at least give me a good area for canned user input.
________________________________

**Reply by alagrede** [reply]
Hi @lostecmo,
To complete my previous answer, I had some useful examples explaining how to load external libs, scripts, manipulate data...
https://github.com/alagrede/znote-app/tree/master/examples
Check the **expressjs** example, I put an HTML input and send user value to a server. Or maybe the **React** example.

I would be very interested to know in detail what you are trying to do. Could you try to explain me your use-case? Maybe I could help you build it and isolate new features to put in the application.

Anthony

**Comment by lostecmo** [comment]
Thank you so much for the update.  I will review and try the examples.  As for what I'm doing... for the most part, trying to learn more javascript/html/etc.  Thought NotebookJS and/or ZNote would be a nice platform to use for it.  Similar to Jupyter for python.  In the long run, I'm hoping to use it as a working notebook for development of some private applications for use in trading various stocks/forex/etc.  So I want to see what I can create for a bit of a user interface / display of information.

Sincerely yours,
Alan
________________________________
